### PR TITLE
Added action for obsidian knife and blessed wooden stake

### PR DESCRIPTION
**Pull request by EPuncker**
- script is the one that was around the internet on various datapacks
- feel free to suggest edits and whatsoever

tested here: https://i.imgur.com/5Kl2XOC.gif

### CODE REVIEW

File: data/scripts/actions/skinning.lua [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+		-- Bone Beast
+		[3031] = {chance = 6000, newItem = 5925, after = 3032},
+		[6030] = {chance = 6000, newItem = 5925, after = 3032}, -- after being killed
+
+		-- Clomp
+		[25399] = {chance = 50000, newItem = 24842, after = 25400},
+		[25398] = {chance = 50000, newItem = 24842, after = 25400}, -- after being killed
+
+		-- Piece of Marble Rock
+		[11343] = { {chance = 530, newItem = 11346, desc = "This little figurine of a goddess was masterfully sculpted by |PLAYERNAME|."},
```
Comment (DSpeichert) [inline data/scripts/actions/skinning.lua]: Indent that further in a new line

File: data/scripts/actions/skinning.lua [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+local skinning = Action()
+
+function skinning.onUse(player, item, fromPosition, target, toPosition, isHotkey)
+	local skin = config[item.itemid][target.itemid]
+	if not skin then
+		player:sendCancelMessage(RETURNVALUE_NOTPOSSIBLE)
+		return true
+	end
+
+	local random, effect, transform = math.random(1, 100000), CONST_ME_MAGIC_GREEN, true
```
Comment (DSpeichert) [inline data/scripts/actions/skinning.lua]: one per line please

**Comment by EPuncker** [comment]
> To be really ocd the vampire fix should probably be its own PR.
> But I don't see an issue with with it being fixed with this merge either.

ikr 😝

File: data/scripts/actions/skinning.lua [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+		return true
+	end
+
+	local randomChance = math.random(1, 100000)
+	local effect = CONST_ME_MAGIC_GREEN
+	local transform = true
+	if type(skin[1]) == "table" then
+		local added = false
+		local _skin
+		for i = 1, #skin do
```
Comment (nekiro) [inline data/scripts/actions/skinning.lua]: iterate with ipairs and remove _skin variable
for _, skinChild in ipairs(skin)

Comment (WibbenZ) [reply data/scripts/actions/skinning.lua]: Numeric loops are faster compared to pairs / ipairs
Ex https://medium.com/@RBX_Crazyman32/battle-of-the-loops-c001bcb4961c#1e2e

Comment (nekiro) [reply data/scripts/actions/skinning.lua]: quoting the thing you linked:
`A “numeric” loop is faster. However, in most applications, using a “generic” loop is fine. Unless you actually have performance issues, don’t worry about switching to “numeric” loops. `

Comment (WibbenZ) [reply data/scripts/actions/skinning.lua]: Yes both are fine but why change it? It won't make the code cleaner or faster etc
How ever I agree to change the variable name.

Comment (EPuncker) [reply data/scripts/actions/skinning.lua]: sorry but my brain lacks the knowledge to do whatever you guys are suggesting xD I'd appreciate a pastebin with the change tho ;P

Comment (WibbenZ) [reply data/scripts/actions/skinning.lua]:
```
		for _, skinChild in ipairs(skin)
			if randomChance <= skinChild.chance then
```

Then change all the _skin with skinChild 

If you ask me changing all the _skin to skinChild is enough.

Comment (EPuncker) [reply data/scripts/actions/skinning.lua]: please check if it is correct now

**Comment by EPuncker** [comment]
> Nothing was about right / wrong in this case.
> Both cases work and do the exact same thing, the prev version you had was abit faster thats all.
> 
> In the future I would like that you create a new PR for smaller changes (like you did here, a small change to a monster and now adding a path change to a file)

okay, will do

**Pull request merged**